tests/integrations/mcs/tso/api_test.go

```diff
@@ -104,3 +109,67 @@ func mustGetKeyspaceGroupMembers(re *require.Assertions, server *tso.Server) map
 	re.NoError(json.Unmarshal(data, &resp))
 	return resp
 }
+
+func TestTSOServerStartFirst(t *testing.T) {
+	re := require.New(t)
+	re.NoError(failpoint.Enable("github.com/tikv/pd/server/delayStartServer", `return(true)`))
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	pdCluster, err := tests.NewTestAPICluster(ctx, 1, func(conf *config.Config, serverName string) {
+		conf.Keyspace.PreAlloc = []string{"k1", "k2"}
+	})
+	defer pdCluster.Destroy()
+	re.NoError(err)
+	addr := pdCluster.GetConfig().GetClientURL()
+	ch := make(chan struct{})
+	defer close(ch)
+	clusterCh := make(chan *mcs.TestTSOCluster)
+	defer close(clusterCh)
+	go func() {
+		tsoCluster, err := mcs.NewTestTSOCluster(ctx, 2, addr)
+		re.NoError(err)
```

Comment:
Could NewTestTSOCluster fail at etcdutil.CreateClients() because pdCluster.RunInitialServers() hasn't started etcd at that moment?

Reply:
No, it will wait for API server to be ready.